components/chunk_text/src/main.py

```diff
@@ -0,0 +1,68 @@
+"""
+Component that chunks text into smaller segments.
+
+This component takes a body of text and chunks into small chunks. The id of the returned dataset
+consists of the id of the original document followed by the chunk index.
+
+"""
+import itertools
+import logging
+import typing as t
+
+import pandas as pd
+from fondant.component import PandasTransformComponent
+from langchain.text_splitter import RecursiveCharacterTextSplitter
+
+logger = logging.getLogger(__name__)
+
+
+def chunk_text(row, text_splitter: RecursiveCharacterTextSplitter) -> t.List[t.Tuple]:
```

Comment:
I would move this function into the `Component`, so we can just access the text splitter as `self.text_splitter` instead of passing it around.